extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java

```diff
@@ -54,26 +55,21 @@
 
     private static final Logger LOGGER = Logger.getLogger(LiquibaseServiceLoader.class);
 
-    /**
-     * The change log parser factory singleton used for the liquibase service classpath scanner
-     */
-    private static final ChangeLogParserFactory CHANGE_LOG_PARSER_FACTORY = ChangeLogParserFactory.getInstance();
-
-    /**
-     * Liquibase build time configuration
-     */
     LiquibaseBuildTimeConfig liquibaseBuildConfig;
 
     @BuildStep
     CapabilityBuildItem capability() {
         return new CapabilityBuildItem(Capabilities.LIQUIBASE);
     }
 
-    @BuildStep
-    void reflection(BuildProducer<ReflectiveClassBuildItem> reflective,
-            BuildProducer<NativeImageResourceBuildItem> resourceProducer,
-            BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer,
-            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized) {
+    @BuildStep(onlyIf = NativeBuild.class, loadsApplicationClasses = true)
```

Comment:
This is the gist of it: only analyzing the changelog in the native image case as we don't need it in other cases.